snapshot-labs/snapshot
# fix(UX): fix keyboard navigation

### Issues
This PR fixed UX issues where keyboard navigation is not intuitive and fluid.

Fixes #3707 

### Changes 

1. Fix most of the nested `<button>` inside `<a>/<button>`, removing the redundant focus when navigating through the links, otherwise, disable the redundant focus by adding `tabindex="-1"` to the element without navigation event
2. Converting `<div>` with onClick event to `<a>` or `<button>`, to make them focusable, whenever possible
3. Make navigation element visually focusable (remove the `outline-none` class)
4. Fix popover link (e.g. TheNavbar component): auto-focus element inside the popover on open

### How to test

1. Open any pages, and navigate through the page with the keyboard with `TAB`
2. The focus should jump from one link to the next logical one

### To-Do
Not everything are fixed, only the most used components are fixed for now.
Some more advanced pages, such as plugins, are not fixed yet.


### Self-review checklist

### Additional notes or considerations
Nothing new is introduced, the worst that could happen are some broken links.

Keyboard navigation is still broken with modal, opening a modal does auto move the next `TAB` focus inside the modal. This is due to the custom modal implementation, which does not implement the necessary [focus helper](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/src/utils/focus-management.ts) found in the headlessui modal

UI-wise, some focus style are not matching the button style (missing the rounded corner), as the focus is on the outer element, without the button style.

There's some unrelate html class sorting changes, automatically done by prettier

There's also some redundancy by setting `@click` and `@keypress`. Is there a way to set both of them at once?
